### PR TITLE
[SPARK-20620][TEST]Improve some unit tests for NullExpressionsSuite and TypeCoercionSuite

### PR DESCRIPTION
**Pull request by 10110346**
## What changes were proposed in this pull request?
add more  datatype for some unit tests

## How was this patch tested?
unit tests

### CODE REVIEW

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/NullExpressionsSuite.scala [hunk 1]
```diff
@@ -87,24 +91,82 @@ class NullExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     }
   }
 
-  test("SPARK-16602 Nvl should support numeric-string cases") {
-    def analyze(expr: Expression): Expression = {
-      val relation = LocalRelation()
-      SimpleAnalyzer.execute(Project(Seq(Alias(expr, "c")()), relation)).expressions.head
-    }
-
+  test("coalesce -check return datatype") {
```
Comment (gatorsmile) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/NullExpressionsSuite.scala]: This test is part of the test case `test("coalesce casts")` in `TypeCoercionSuite`. If you want, you can improve that test case.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/NullExpressionsSuite.scala [hunk 2]
```diff
@@ -87,24 +91,82 @@ class NullExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
...
     assert(analyze(new Nvl(nullLit, intLit)).dataType == IntegerType)
     assert(analyze(new Nvl(doubleLit, nullLit)).dataType == DoubleType)
     assert(analyze(new Nvl(nullLit, stringLit)).dataType == StringType)
+
+    assert(analyze(new Nvl(floatLit, stringLit)).dataType == StringType)
+    assert(analyze(new Nvl(floatLit, doubleLit)).dataType == DoubleType)
+    assert(analyze(new Nvl(floatNullLit, intLit)).dataType == FloatType)
+  }
+
+  test("Nvl2 -check return datatype") {
```
Comment (gatorsmile) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/NullExpressionsSuite.scala]: The same for this. It is being covered by the test case `test("type coercion for If")` in `TypeCoercionSuite`. If you want, you can improve that test case.

**Comment by 10110346** [comment]
@gatorsmile thanks,l will do it.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercionSuite.scala [hunk 3]
```diff
@@ -484,24 +484,50 @@ class TypeCoercionSuite extends PlanTest {
...
-        :: Literal(new java.math.BigDecimal("1000000000000000000000"))
-        :: Nil),
-      Coalesce(Cast(Literal(1L), DecimalType(22, 0))
-        :: Cast(Literal(1), DecimalType(22, 0))
-        :: Cast(Literal(new java.math.BigDecimal("1000000000000000000000")), DecimalType(22, 0))
-        :: Nil))
+    val rule = TypeCoercion.FunctionArgumentConversion
+
+    val intLit = Literal(1)
+    val LongLit = Literal.create(1L)
```
Comment (HyukjinKwon) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercionSuite.scala]: (I guess this should be `longLit `)

**Comment by 10110346** [comment]
done,thanks @HyukjinKwon

**Comment by 10110346** [comment]
I have modify `Scala style`.
Test is not started, could you help trigger it,thanks  @HyukjinKwon @gatorsmile

**Comment by gatorsmile** [comment]
retest this please

**Comment by gatorsmile** [comment]
LGTM pending Jenkins.

**Comment by 10110346** [comment]
Test passed, thanks.  @gatorsmile

**Comment by gatorsmile** [comment]
retest this please

**Pull request closed without merging**